### PR TITLE
Added Focus Search on App Start

### LINKED ISSUE #821: Add setting to focus search on app start
It would be awesome for quick access if we could have a setting functionally equivalent to [andOTP's "Focus search on app start"](https://github.com/andOTP/andOTP/pull/658) where immediately upon unlocking the vault, the search bar is focused, and the keyboard displayed for the user to search for items

### PR DESCRIPTION
**Pull request by Hacker437**
This PR Will resolve #821

### CODE REVIEW

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java [hunk 1]
```diff
@@ -768,7 +777,7 @@ public void onLocked(boolean userInitiated) {
         if (userInitiated) {
             startAuthActivity(true);
         } else {
-            super.onLocked(userInitiated);
+            super.onLocked(false);
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:780]: Why was this changed?

File: app/src/main/res/xml/preferences_behavior.xml [hunk 2]
```diff
@@ -2,6 +2,12 @@
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:title="@string/pref_section_behavior_title">
+    <androidx.preference.SwitchPreferenceCompat
+        android:defaultValue="false"
+        android:key="pref_focus"
```
Comment (alexbakker) [inline app/src/main/res/xml/preferences_behavior.xml:7]: Let's rename this to ``pref_focus_search``.

File: app/src/main/java/com/beemdevelopment/aegis/ui/fragments/BehaviorPreferencesFragment.java [hunk 3]
```diff
@@ -19,6 +19,12 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
 
         _prefs = getPreferences();
 
+        Preference focusSearchPreference = findPreference("pref_focus");
+        focusSearchPreference.setOnPreferenceChangeListener((preference, newValue) -> {
+            getResult().putExtra("needsRefresh", true);
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/fragments/BehaviorPreferencesFragment.java:24]: Triggering a refresh should not be needed here.

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java [hunk 4]
```diff
@@ -561,8 +564,14 @@ public boolean onCreateOptionsMenu(Menu menu) {
         MenuItem searchViewMenuItem = menu.findItem(R.id.mi_search);
 
         _searchView = (SearchView) searchViewMenuItem.getActionView();
-        _searchView.setFocusable(false);
+
         _searchView.setQueryHint(getString(R.string.search));
+        if(getPreferences().getFocusSearchEnabled() && !_recreate) {
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:569]: Missing a space after ``if`` here.

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java [hunk 5]
```diff
@@ -81,6 +81,7 @@
 
     private boolean _isAuthenticating;
     private boolean _isDoingIntro;
+    private boolean _recreate;
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:84]: Please rename this to ``_isRecreated``.

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java [hunk 6]
```diff
@@ -105,6 +106,7 @@ protected void onCreate(Bundle savedInstanceState) {
         _loaded = false;
 
         if (savedInstanceState != null) {
+            _recreate = savedInstanceState.getBoolean("isReCreate");
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:109]: Do we really need to store this in the instance state bundle, or can we just directly assign ``true`` here? That seems equivalent and shorter.

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java [hunk 7]
```diff
@@ -561,8 +564,14 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
 
         _searchView = (SearchView) searchViewMenuItem.getActionView();
-        _searchView.setFocusable(false);
+
         _searchView.setQueryHint(getString(R.string.search));
+        if(getPreferences().getFocusSearchEnabled() && !_recreate) {
+            _searchView.setIconified(false);
+            _searchView.setFocusable(true);
+            _searchView.requestFocus();
+            _searchView.requestFocusFromTouch();
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:573]: Why is there a call to ``requestFocusFromTouch`` here? Is ``requestFocus`` not enough?

Why is ``setFocusable(true)`` called here? Is it ever false?

**Review by alexbakker — CHANGES_REQUESTED** [review]
Functionally this looks alright, but I have a couple of comments.

File: app/src/main/res/xml/preferences_behavior.xml (hunk 2 above)
Comment (Hacker437) [reply app/src/main/res/xml/preferences_behavior.xml:7]: Sure

File: app/src/main/java/com/beemdevelopment/aegis/ui/fragments/BehaviorPreferencesFragment.java (hunk 3 above)
Comment (Hacker437) [reply app/src/main/java/com/beemdevelopment/aegis/ui/fragments/BehaviorPreferencesFragment.java:24]: Forgot to remove it

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java (hunk 1 above)
Comment (Hacker437) [reply app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:780]: Because it always is false there , Because if userInitiated is false it calls that super and it is not related to this PR, Optional

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java (hunk 5 above)
Comment (Hacker437) [reply app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:84]: Sure

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java (hunk 4 above)
Comment (Hacker437) [reply app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:569]: Ok

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java (hunk 7 above)
Comment (Hacker437) [reply app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:573]: requestFocusFromTouch is not needed.
Better to call setFocuasable true

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java (hunk 6 above)
Comment (Hacker437) [reply app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:109]: Enough i didn't think that much.

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java (hunk 5 above)
Comment (Hacker437) [reply app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:84]: Sure

**Comment by Hacker437** [comment]
Made the requested changes

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java (hunk 1 above)
Comment (alexbakker) [reply app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:779]: You're right. This is fine for now, but I'd prefer it if you could separate any unrelated changes in the future.

**Review by alexbakker — APPROVED** [review]
Thanks!

**Comment by Hacker437** [comment]
> Thanks!

@alexbakker Happy to help

**Comment by Hacker437** [comment]
@alexbakker There is a problem with I think had forgot to add the setting to manually chat it!

**Comment by alexbakker** [comment]
@Hacker437 I'm not sure what you mean. Could you elaborate?

**Comment by Hacker437** [comment]
@alexbakker I have fogot to commit the Behavior Preference Setting file Which means that There is no way to change the "Focus app on Start" Setting manually

**Comment by alexbakker** [comment]
@Hacker437 I'm still not sure what you mean. I can see the "Focus search on app start" under "Behavior", and changing it has the expected affect.

**Comment by Hacker437** [comment]
> @Hacker437 I'm still not sure what you mean. I can see the "Focus search on app start" under "Behavior", and changing it has the expected affect.

@alexbakker Oops sorry, Forgot to use git pull 😅😅
